test/unit/workers/document_list_export_worker_test.rb

```diff
@@ -22,12 +41,41 @@ class DocumentListExportWorkerTest < ActiveSupport::TestCase
     @worker.perform({ "state" => "draft" }, @user.id)
   end
 
-  test "sends mail with CSV to user" do
+  test "uploads csv to s3" do
+    csv = "this|is|my|csv"
+    title = "Everyone's editions"
+    @worker.stubs(:create_filter).returns(stub(page_title: title))
+    @worker.stubs(generate_csv: csv)
+    @worker.stubs(:send_mail)
+
+    @worker.perform({ "state" => "draft" }, @user.id)
+
+    assert_equal 1, @directory.files.length
+    file = @directory.files[0]
+    assert_not_nil file
+    assert_match %r{^document_list_documents_[a-f0-9-]{36}\.csv$}, file.key
+    assert_equal "this|is|my|csv", file.body
+  end
+
+  test "email body includes url" do
+    csv = "this|is|my|csv"
+    title = "Everyone's editions"
+    @worker.stubs(:create_filter).returns(stub(page_title: title))
+    @worker.stubs(generate_csv: csv)
+    # @worker.stubs(:send_mail)
+
+    @worker.perform({ "state" => "draft" }, @user.id)
+
+    message = ActionMailer::Base.deliveries.last
+    assert_match %r{\bhttps://whitehall\.test\.gov\.uk/export/documents/[a-f0-9-]{36}\b}, message.body.raw_source
+  end
+
+  test "sends mail with CSV link to user" do
     csv = "this|is|my|csv"
     title = "Everyone's editions"
     @worker.stubs(:create_filter).returns(stub(page_title: title))
     @worker.stubs(generate_csv: csv)
-    Notifications.expects(:document_list).with(csv, @user.email, title).returns(stub(deliver_now: nil))
+    Notifications.expects(:document_list).with(regexp_matches(%r{^https://whitehall\.test\.gov\.uk/export/documents/[a-f0-9-]{36}$}), @user.email, title).returns(stub(deliver_now: nil))
```

Comment:
I'd be tempted to extract this regex into a variable/method so we avoid repeating it twice.

Reply:
One of the instances has been removed but it's still used in a couple of different places — it could go in `test_helper.rb` but it feels a little bit too specific for that.